src/image_builder.py

```diff
@@ -154,109 +155,177 @@ def image_builder(buildspec):
             "labels": labels,
             "extra_build_args": extra_build_args
         }
-
-        image_object = DockerImage(
+        
+        #Create first stage docker object
+        first_stage_image_object = DockerImage(
             info=info,
             dockerfile=image_config["docker_file"],
             repository=image_repo_uri,
             tag=image_tag,
             to_build=image_config["build"],
+            stage=constants.FIRST_STAGE,
             context=context,
         )
 
-        IMAGES.append(image_object)
-
-    FORMATTER.banner("DLC")
-    FORMATTER.title("Status")
-
-    THREADS = {}
+        #Create second stage docker object
+        if "example" not in image_name.lower() and build_context == "MAINLINE":
+            second_stage_image_object = DockerImage(
+                info=info,
+                dockerfile=os.path.join(os.sep, "Dockerfile.multipart"),
+                repository=image_repo_uri,
+                tag=image_tag,
+                to_build=image_config["build"],
+                stage=constants.SECOND_STAGE,
+                context=None,
+            )
 
-    # In the context of the ThreadPoolExecutor each instance of image.build submitted
-    # to it is executed concurrently in a separate thread.
-    with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
-        # Standard images must be built before example images
-        # Example images will use standard images as base
-        standard_images = [image for image in IMAGES if "example" not in image.name.lower()]
-        example_images = [image for image in IMAGES if "example" in image.name.lower()]
+        FORMATTER.separator()
 
-        for image in standard_images:
-            THREADS[image.name] = executor.submit(image.build)
+        FIRST_STAGES_IMAGES.append(first_stage_image_object)
+        SECOND_STAGES_IMAGES.append(second_stage_image_object)
```

Comment:
This line would fail if the above `if` statement is not true. Would say something like `local variable 'second_stage_image_object' referenced before assignment`